sdks/RepoToolset/tools/Compiler.props

```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Net.Compilers" Version="$(MicrosoftNetCompilersVersion)" />
+    <PackageReference Include="Microsoft.Net.Compilers" Version="$(MicrosoftNetCompilersVersion)" PrivateAssets="all" />
```

Comment:
Can we elaborate on what was breaking here? I agree this is the right choice but also want to understand what impact this had (can't think of a prob it creates).

Reply:
`Microsoft.Net.Compilers` is a development dependency, but `PackageReference` doesn't have support for `developmentDependency` metadata yet. Currently, packages created from project files (like `Microsoft.VisualStudio.ProjectSystem.Managed`) pull `Microsoft.Net.Compilers`, but it shouldn't be included as a package dependency, as it's only used to build the projects.